lib/src/migrators/module/scope.dart

```diff
@@ -0,0 +1,140 @@
+  /// they can't be migrated to @use.
+  final nestedImports = <Uri>{};
+
+  Scope(this.parent);
+
+  /// The global scope this scope descends from.
+  Scope get global => parent?.global ?? this;
+
+  /// Returns whether a variable [name] exists in a non-global scope.
+  bool isLocalVariable(String name) =>
+      parent != null &&
+      (variables.containsKey(name) || parent.isLocalVariable(name));
+
+  /// Returns whether a mixin [name] exists in a non-global scope.
+  bool isLocalMixin(String name) =>
+      parent != null && (mixins.containsKey(name) || parent.isLocalMixin(name));
+
+  /// Returns whether a function [name] exists in a non-global scope.
+  bool isLocalFunction(String name) =>
+      parent != null &&
+      (functions.containsKey(name) || parent.isLocalFunction(name));
+
+  /// Constructs and throws a MigrationException for a reference to a member
+  /// from a nested import.
+  void _errorNestedImport(String type, Uri importUrl, FileSpan reference) {
+    throw MigrationException(
+        "Can't reference $type from ${p.prettyUri(importUrl)}, "
+        "as load-css() does not load ${type}s.",
+        span: reference);
+  }
+
+  /// Checks that [node] does not reference a variable from a nested import,
+  /// throwing a MigrationException if it does.
+  void checkNestedImportVariable(VariableExpression node) {
+    if (variables.containsKey(node.name)) {
+      var url = variables[node.name].span.sourceUrl;
+      if (nestedImports.contains(url)) {
+        _errorNestedImport('variable', url, node.span);
+      }
+    } else {
+      parent?.checkNestedImportVariable(node);
+    }
+  }
+
+  /// Checks that [node] does not reference a mixin from a nested import,
+  /// throwing a MigrationException if it does.
+  void checkNestedImportMixin(IncludeRule node) {
+    if (mixins.containsKey(node.name)) {
+      var url = mixins[node.name].span.sourceUrl;
+      if (nestedImports.contains(url)) {
+        _errorNestedImport('mixin', url, node.span);
+      }
+    } else {
+      parent?.checkNestedImportMixin(node);
+    }
+  }
+
+  /// Checks that [node] does not reference a function from a nested import,
+  /// throwing a MigrationException if it does.
+  void checkNestedImportFunction(FunctionExpression node) {
+    if (node.name.asPlain == null) return;
+    if (functions.containsKey(node.name.asPlain)) {
+      var url = functions[node.name.asPlain].span.sourceUrl;
+      if (nestedImports.contains(url)) {
+        _errorNestedImport('function', url, node.span);
+      }
+    } else {
+      parent?.checkNestedImportFunction(node);
+    }
+  }
+
+  /// Create a flattened version of this scope, combining all members from all
+  /// ancestors into one level.
+  ///
+  /// This is used for migrating nested imports, allowing the migrator to treat
+  /// local members from the upstream stylesheet as global ones while migrating
+  /// the nested import.
```

Comment:
Is this desirable? Won't it produce an incorrect migration if a local member is referred to by the nested import? (That's a case that *can't* be migrated, so I'd expect an error.)